.circleci/config.yml

```diff
@@ -61,6 +74,15 @@ commands:
           command: ./.circleci/fail_fast.sh
 
 jobs:
+  'Setup ENV':
+    executor:
+      name: amphtml-medium-executor
+    steps:
+      - checkout
+      - run:
+          name: 'Fetch Merge Commit'
+          command: ./.circleci/fetch_merge_commit.sh fetch
+      - save_merge_commit
```

Comment:
Right now, we're doing a full checkout before saving the merge commit. This adds ~2 mins to the total CI round trip time (1.x mins for running the new job + a few seconds of waiting after the newly added stage).

![image](https://user-images.githubusercontent.com/26553114/110554820-802c2e80-8109-11eb-8d81-f40801970244.png)

It's also resulting in us hitting our account's concurrency limit.

<img width="1439" alt="Screen Shot 2021-03-09 at 7 03 38 PM" src="https://user-images.githubusercontent.com/26553114/110555235-35f77d00-810a-11eb-9f98-bac2e50ad824.png">

At the minimum, I think this step can be done more efficiently without doing a full checkout. Something like this works without any checking out of code:

```sh
$ git ls-remote https://github.com/ampproject/amphtml.git refs/pull/33165/merge
e36e2e05be16fe93ed41846827d14f4259d38cd6        refs/pull/33165/merge
```

A higher-level thought is that the problem of synchronizing baseline commits mainly applies to just those jobs that depend on downloading previously built artifacts, so I think we might be able to do away with this extra stage and just persist the commit at the time of building, and use that as the baseline while reporting statuses. Lemme do some homework on this to see if I can come up with a solution that works.

Reply:
> `git ls-remote https://github.com/ampproject/amphtml.git refs/pull/33165/merge`

HOW DID YOU FIND THIS?!

> A higher-level thought is that the problem of synchronizing baseline commits mainly applies to just those jobs that depend on downloading previously built artifacts, so I think we might be able to do away with this extra stage and just persist the commit at the time of building

My real desire is to be able to persist the `.git` directory entirely, as in https://circleci.com/docs/2.0/caching/#source-caching. But we need to do https://github.com/ampproject/amphtml/issues/21691 first, so that the ridiculous 800mb of disk space is much smaller. Also, could attempt doing the full clone and commit log list in the Setup job, and let all other jobs do a shallow clone with the SHAs the Setup finds.

Reply:
> My real desire is to be able to persist the `.git` directory entirely, as in [circleci.com/docs/2.0/caching/#source-caching](https://circleci.com/docs/2.0/caching/#source-caching).

Interesting.

> But we need to do #21691 first, so that the ridiculous 800mb of disk space is much smaller. 

We did use LFS for https://github.com/ampproject/amp-closure-compiler in the past, but quickly ran out of LFS quota for the `ampproject` enterprise GH account. If we try using LFS for `amphtml`, I'm sure we'll need to create a new payment plan.

![image](https://user-images.githubusercontent.com/26553114/110561860-71984400-8116-11eb-8040-60e5f15fa015.png)

> Also, could attempt doing the full clone and commit log list in the Setup job, and let all other jobs do a shallow clone with the SHAs the Setup finds.

Looks like the combination of docker and `git ls-remote` worked out better.